sdk/keyvault/Azure.Security.KeyVault.Administration/tests/ChallengeBasedAuthenticationPolicyTests.cs

```diff
@@ -10,39 +12,85 @@
 
 namespace Azure.Security.KeyVault.Tests
 {
+    [NonParallelizable]
     public class ChallengeBasedAuthenticationPolicyTests : SyncAsyncPolicyTestBase
     {
         internal ChallengeBasedAuthenticationPolicy _policy;
         private const string KeyVaultChallenge = "Bearer authorization=\"https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47\", resource=\"https://vault.azure.net\"";
         public ChallengeBasedAuthenticationPolicyTests(bool isAsync) : base(isAsync)
         {
-            _policy = new ChallengeBasedAuthenticationPolicy(new MockCredentialThrowsWithNoScopes());
+            _policy = new ChallengeBasedAuthenticationPolicy(new MockCredentialThrowsWithNoScopes(), true);
         }
 
-        [Test]
-        [NonParallelizable]
-        public async Task ScopesAreInitializedFromCache()
+        [SetUp]
+        public void SetUp()
         {
             // Clear the cache to ensure the test starts with an empty cache.
             ChallengeBasedAuthenticationPolicy.ClearCache();
+        }
 
+        [Test]
+        public async Task ScopesAreInitializedFromCache()
+        {
             var keyvaultChallengeResponse = new MockResponse(401);
             keyvaultChallengeResponse.AddHeader(new HttpHeader("WWW-Authenticate", KeyVaultChallenge));
             MockTransport transport = CreateMockTransport(keyvaultChallengeResponse, new MockResponse(200));
 
-            var response = await SendGetRequest(transport, _policy, uri: new Uri("https://example.com"));
+            Response response = await SendGetRequest(transport, _policy, uri: new Uri("https://myvault.vault.azure.net"));
 
             Assert.That(response.Status, Is.EqualTo(200));
 
             // Construct a new policy so that we can get the Scopes from cache.
```

Comment:
I am a little confused by this test case. How are we validating here that the scope is getting picked from the  cache in the second call to sendgetRequest. Challengebasedauthpolicy may have it cached the first time but the second sendGetRequest will re add it if it were not present. Did you mean to force the check even if scope was not present (via cache) and accidently used the same request the second time?

Reply:
The cache is static, and because I only added one challenge mock response, if it weren't cache the client (via the new policy) would've expected a WWW-Authenticate header and failed since one wasn't queued by the `MockTransport` (responses are popped as they are used unless you use the delegate form constructor).

Reply:
I may be mistaken here but line 46 has the following  transport = CreateMockTransport(keyvaultChallengeResponse, new MockResponse(200)); --> which is 2 responses. Did you intent to only add one response to the transport instead of 2?

Reply:
Sorry, you're right! Must've gotten a little overzealous on copy/paste. Fix incoming.